src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -288,6 +289,17 @@ private IReadOnlyList<SemanticRange> CombineSemanticRanges(params IReadOnlyList<
 
             if (csharpResponse is null || csharpResponse.HostDocumentSyncVersion != documentVersion)
             {
+                if (csharpResponse != null && csharpResponse.HostDocumentSyncVersion is null)
```

Comment:
This is how I'm identifying if the result is from VSCode, though it would be preferable to fix this by having the serializer just not tell lies.

Reply:
O_o wait so why does the deserializer fall over?